Presentation/UINavigationController+Presenting.swift

```diff
@@ -176,7 +176,14 @@ private extension UINavigationController {
                 setViewControllers(vcs, animated: animated)
             }
         } else {
-            setViewControllers(vcs, animated: animated)
+            // push and pop animations doesn't work for some reason with setViewControllers, so defaulting to pushViewController and popViewController
+            if vcs.count == viewControllers.count + 1 {
+                pushViewController(vcs.last!, animated: animated)
+            } else if vcs.count == viewControllers.count - 1 {
```

Comment:
These two checks assumes that the rest of the view controllers in `vcs` and `viewControllers` are the same. I think we can make this check more correct with something like this instead?:

```swift
if vcs.elementsEqual(viewControllers.dropLast()) {
  pushViewController(vcs.last!, animated: animated)
} else if vcs.dropLast().elementsEqual(viewControllers.count) {
```

Reply:
@mansbernhardt oh, that's way better! I've updated the PR!